src/basic/selinux-util.c

```diff
@@ -103,37 +108,98 @@ int mac_selinux_init(void) {
                   format_timespan(timespan, sizeof(timespan), after_timestamp - before_timestamp, 0),
                   (l+1023)/1024);
 
+        return 0;
+}
+#endif
+
+int mac_selinux_init(void) {
+#if HAVE_SELINUX
+        int r;
+
+        if (initialzed)
+                return 0;
+
+        if (!mac_selinux_use())
+                return 0;
+
+        r = selinux_status_open(/* no netlink fallback */ 0);
+        if (r < 0) {
+                selinux_status_page_failed = true;
+                return log_enforcing_errno(errno, "Failed to open SELinux status page: %m");
+        }
+        selinux_status_page_failed = false;
+
+        last_policyload = selinux_status_policyload();
+        /* ignore retrieval failure */
+        if (last_policyload < 0)
+                last_policyload = 0;
+
+        r = open_label_db(&label_hnd);
+        if (r < 0) {
+                selinux_status_close();
+                return r;
+        }
+
+        initialzed = true;
 #endif
         return 0;
 }
 
+void mac_selinux_check_reload_and_update(void) {
+#if HAVE_SELINUX
+        int r;
+
+        r = selinux_status_updated();
+        if (r < 0)
+                log_debug_errno(errno, "Failed to update SELinux from status page: %m");
+        if (r > 0) {
+                int policyload;
+
+                log_debug("SELinux status page update");
+
+                /* from libselinux > 3.1 callbacks gets automatically called, see
+                   https://github.com/SELinuxProject/selinux/commit/05bdc03130d741e53e1fb45a958d0a2c184be503 */
+
+                /* only reload on policy changes, not enforcing status changes */
+                policyload = selinux_status_policyload();
+                if (policyload != last_policyload) {
+                        mac_selinux_reload(policyload);
+                        /* do not update on retrieval failure */
+                        if (policyload >= 0)
+                                last_policyload = policyload;
+                }
+        }
+#endif
```

Comment:
why are you eating up errors here?

Reply:
dropped.
`selinux_status_policyload()` only fails if `selinux_status_open()` wasn't succesful, but if that's the case `selinux_status_updated()` will also have failed.